src/components/ha-code-editor.ts

```diff
@@ -123,6 +123,7 @@ export class HaCodeEditor extends ReactiveElement {
     }
     const transactions: TransactionSpec[] = [];
     if (changedProps.has("mode")) {
+      // TODO: not sure how to handle things here
```

Comment:
left this as a note so I wouldn't forget. I probably need some guidance here. Not sure if/how to handle mode changes
can mode changes even occur at all? I assume we need to check if mode changed to/from yaml, and disable/enable the added extensions accordingly, but not sure how to go about it

Reply:
You would need to make it a compartment, so you can reconfigure it.

Reply:
got it. will take a look at that, thanks